### ISSUE TITLE
AttributeError: 'StableDiffusionProcessingImg2Img' object has no attribute 'iteration'

### ISSUE THREAD

**Issue by harlyh** [AlUlkesh/sd_save_intermediate_images#22]
As long as you don't have this:
![image](https://user-images.githubusercontent.com/30772586/217308442-dd41cdc2-6e38-4f44-91b0-225df2c70ec8.png)
checked, you're good. 

Bur as soon as you do check it, you will get the below Exception that you can't escape even after unchecking.
```
Traceback (most recent call last):
  File "E:\AI\stable-diffusion-webui\modules\call_queue.py", line 45, in f
    res = list(func(*args, **kwargs))
  File "E:\AI\stable-diffusion-webui\modules\call_queue.py", line 28, in f
    res = func(*args, **kwargs)
  File "E:\AI\stable-diffusion-webui\modules\img2img.py", line 152, in img2img
    processed = process_images(p)
  File "E:\AI\stable-diffusion-webui\modules\processing.py", line 470, in process_images
    res = process_images_inner(p)
  File "E:\AI\stable-diffusion-webui\modules\processing.py", line 575, in process_images_inner
    samples_ddim = p.sample(conditioning=c, unconditional_conditioning=uc, seeds=seeds, subseeds=subseeds, subseed_strength=p.subseed_strength, prompts=prompts)
  File "E:\AI\stable-diffusion-webui\modules\processing.py", line 917, in sample
    samples = self.sampler.sample_img2img(self, self.init_latent, x, conditioning, unconditional_conditioning, image_conditioning=self.image_conditioning)
  File "E:\AI\stable-diffusion-webui\modules\sd_samplers.py", line 501, in sample_img2img
    samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args={
  File "E:\AI\stable-diffusion-webui\modules\sd_samplers.py", line 439, in launch_sampling
    return func()
  File "E:\AI\stable-diffusion-webui\modules\sd_samplers.py", line 501, in <lambda>
    samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args={
  File "E:\AI\stable-diffusion-webui\venv\lib\site-packages\torch\autograd\grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "E:\AI\stable-diffusion-webui\repositories\k-diffusion\k_diffusion\sampling.py", line 148, in sample_euler_ancestral
    callback({'x': x, 'i': i, 'sigma': sigmas[i], 'sigma_hat': sigmas[i], 'denoised': denoised})
  File "E:\AI\stable-diffusion-webui\extensions\sd_save_intermediate_images\scripts\sd_save_intermediate_images.py", line 455, in callback_state
    p.intermed_batch_iter = p.iteration
AttributeError: 'StableDiffusionProcessingImg2Img' object has no attribute 'iteration'
```
In addition, after the above exception, clicking this:
![image](https://user-images.githubusercontent.com/30772586/217307677-9fe31138-e735-4cf2-bb4d-768f995e52a7.png)
will end with this:

```
Traceback (most recent call last):
  File "E:\AI\stable-diffusion-webui\venv\lib\site-packages\gradio\routes.py", line 284, in run_predict
    output = await app.blocks.process_api(
  File "E:\AI\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 982, in process_api
    result = await self.call_function(fn_index, inputs, iterator)
  File "E:\AI\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 824, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "E:\AI\stable-diffusion-webui\venv\lib\site-packages\anyio\to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "E:\AI\stable-diffusion-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "E:\AI\stable-diffusion-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "E:\AI\stable-diffusion-webui\extensions\sd_save_intermediate_images\scripts\sd_save_intermediate_images.py", line 74, in ssii_save_settings_do
    ui_settings = ui_setting_set(ui_settings, value, eval(key))
  File "E:\AI\stable-diffusion-webui\extensions\sd_save_intermediate_images\scripts\sd_save_intermediate_images.py", line 51, in ui_setting_set
    this_module = os.path.basename(__file__)
NameError: name '__file__' is not defined
```

I hope this helps.

**Comment by harlyh** [comment]
OK, So, after I identified and fixed the issue I realised that my WebUi doesn't have fixes from #20 in it

**Comment by harlyh** [comment]
OK, I realised that another package (steps_animation) needs the same fix as your #20 did.
